pints/tests/test_residuals_diagnostics.py

```diff
@@ -0,0 +1,199 @@
+#
+# Test the methods in pints.residuals_diagnostics
+#
+# This file is part of PINTS
+#  Copyright (c) 2017-2019, University of Oxford.
+#  For licensing informating, see the LICENSE file distributed with the PINTS
+#  software package.
+#
+import pints
+import pints.residuals_diagnostics
+import pints.toy as toy
+import unittest
+import numpy as np
+import matplotlib
+
+# Select matplotlib backend that doesn't require a screen
+matplotlib.use('Agg')
```

Comment:
This is an example of the above! If you had an `import matplotlib.pyplot` anywhere above this line (including inside pints), matplotlib would complain

(This code is fine, by the way, just continuing my above comment :-) )